compiler/src/dotty/tools/dotc/typer/TyperPhase.scala

```diff
@@ -28,43 +29,14 @@ class FrontEnd extends Phase {
   /** The contexts for compilation units that are parsed but not yet entered */
   private var remaining: List[Context] = Nil
 
-  /** The position of the first XML literal encountered while parsing,
-   *  NoSourcePosition if there were no XML literals.
-   */
-  private var firstXmlPos: SourcePosition = NoSourcePosition
-
   /** Does a source file ending with `<name>.scala` belong to a compilation unit
    *  that is parsed but not yet entered?
    */
   def stillToBeEntered(name: String): Boolean =
     remaining.exists(_.compilationUnit.toString.endsWith(name + ".scala"))
 
-  def monitor(doing: String)(body: => Unit)(using Context): Unit =
-    try body
-    catch
-      case NonFatal(ex) =>
-        report.echo(s"exception occurred while $doing ${ctx.compilationUnit}")
-        throw ex
-
-  def parse(using Context): Unit = monitor("parsing") {
-    val unit = ctx.compilationUnit
-
-    unit.untpdTree =
-      if (unit.isJava) new JavaParser(unit.source).parse()
-      else {
-        val p = new Parser(unit.source)
-       //  p.in.debugTokenStream = true
-        val tree = p.parse()
-        if (p.firstXmlPos.exists && !firstXmlPos.exists)
-          firstXmlPos = p.firstXmlPos
-        tree
-      }
-
-    val printer = if (ctx.settings.Xprint.value.contains("parser")) default else typr
-    printer.println("parsed:\n" + unit.untpdTree.show)
-    if (Config.checkPositions)
-      unit.untpdTree.checkPos(nonOverlapping = !unit.isJava && !ctx.reporter.hasErrors)
-  }
+  // Run regardless of parsing errors
+  override def isRunnable(implicit ctx: Context): Boolean = true
```

Comment:
This seems wrong, no?  Then again no checkfile has failed so perhaps I'm wrong.

Reply:
I lifted that from the original PR. Without it, at least `dotty.tools.dotc.CompilationTests` fail.

Reply:
Yes, even in case of errors in parsing, we run the typer phase, this is particularly useful for IDEs since it means the presentation compiler is usable even when your code doesn't typecheck.